collectors/ebpf.plugin/ebpf_mount.c

```diff
@@ -0,0 +1,257 @@
+
+    write_err_chart(NETDATA_EBPF_MOUNT_ERRORS, NETDATA_EBPF_MOUNT_GLOBAL_FAMILY,
+                    mount_publish_aggregated, NETDATA_EBPF_MOUNT_SYSCALL);
+}
+
+/**
+* Main loop for this collector.
+*/
+static void mount_collector(ebpf_module_t *em)
+{
+    mount_threads.thread = mallocz(sizeof(netdata_thread_t));
+    mount_threads.start_routine = ebpf_mount_read_hash;
+    memset(mount_hash_values, 0, sizeof(mount_hash_values));
+
+    mount_values = callocz((size_t)ebpf_nprocs, sizeof(netdata_idx_t));
+
+    netdata_thread_create(mount_threads.thread, mount_threads.name, NETDATA_THREAD_OPTION_JOINABLE,
+                          ebpf_mount_read_hash, em);
+
+    while (!close_ebpf_plugin) {
+        pthread_mutex_lock(&collect_data_mutex);
+        pthread_cond_wait(&collect_data_cond_var, &collect_data_mutex);
+
+        pthread_mutex_lock(&lock);
+
+        ebpf_mount_send_data();
+
+        pthread_mutex_unlock(&lock);
+        pthread_mutex_unlock(&collect_data_mutex);
+    }
+}
+
+/*****************************************************************
+ *
+ *  INITIALIZE THREAD
+ *
+ *****************************************************************/
+
+/**
+ * Create mount charts
+ *
+ * Call ebpf_create_chart to create the charts for the collector.
+ */
+static void ebpf_create_mount_charts()
+{
+    ebpf_create_chart(NETDATA_EBPF_MOUNT_GLOBAL_FAMILY, NETDATA_EBPF_MOUNT_CALLS,
+                      "Calls to mount and umount syscalls.",
+                      EBPF_COMMON_DIMENSION_CALL, NETDATA_EBPF_MOUNT_FAMILY,
+                      NULL,
+                      NETDATA_EBPF_CHART_TYPE_LINE,
+                      NETDATA_CHART_PRIO_EBPF_MOUNT_CHARTS,
+                      ebpf_create_global_dimension,
+                      mount_publish_aggregated, NETDATA_EBPF_MOUNT_SYSCALL);
+
+    ebpf_create_chart(NETDATA_EBPF_MOUNT_GLOBAL_FAMILY, NETDATA_EBPF_MOUNT_ERRORS,
+                      "Errors to mount and umount syscalls.",
+                      EBPF_COMMON_DIMENSION_CALL, NETDATA_EBPF_MOUNT_FAMILY,
+                      NULL,
+                      NETDATA_EBPF_CHART_TYPE_LINE,
+                      NETDATA_CHART_PRIO_EBPF_MOUNT_CHARTS + 1,
+                      ebpf_create_global_dimension,
+                      mount_publish_aggregated, NETDATA_EBPF_MOUNT_SYSCALL);
+
+    fflush(stdout);
+}
+
+/*****************************************************************
+ *
+ *  MAIN THREAD
+ *
+ *****************************************************************/
+
+/**
+ * Directory Cache thread
+ *
+ * Thread used to make dcstat thread
+ *
+ * @param ptr a pointer to `struct ebpf_module`
+ *
+ * @return It always returns NULL
+ */
+void *ebpf_mount_thread(void *ptr)
+{
+    netdata_thread_cleanup_push(ebpf_mount_cleanup, ptr);
+
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+    em->maps = mount_maps;
+    fill_ebpf_data(&mount_data);
+
+    if (!em->enabled)
+        goto endmount;
+
+    if (ebpf_update_kernel(&mount_data))
```

Comment:
Not sure how important this is to note but, `ebpf_update_kernel` always returns 0. We may have discussed this before, but I assume this function is gonna change later? (As it currently is, it seems to just overwrite memory in `mount_data.kernel_string`, which points to a global shared 64-byte array - so all ebpf init threads are writing over each other here).

Reply:
This function will be removed soon, I am using it here, because we need update kernel version before to load, but your conclusions are correct.